### PR TITLE
add error handlers, middlewares, csv import and storing

### PR DESCRIPTION
**Pull request by Kitsune1904**

### CODE REVIEW

File: helpers/helpers.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+    if (productsStore.length > 0) {
+        const isProdExist = productsStore.findIndex(prod => prod.name === appendData.name);
+        if (isProdExist !== -1) {
+            throw new ApiError(409, 'Product exist')
+        }
+    }
+
+    productsStore.push({
+        id: productsStore.length,
```
Comment (VitalyaIlin) [inline helpers/helpers.js:32]: use UUID instead of product counter

File: shop/services.js [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+import bcrypt from "bcrypt";
+import {ApiError} from "../helpers/ErrorApi.js";
+import { userValidationSchema} from "../helpers/validation.js";
+import { EventEmitter } from 'events';
+import fs, { unlink } from 'fs/promises';
+import * as path from "path";
+import {createReadStream, appendFileSync} from "fs";
+import csv from "csv-parser";
+
+const PRODUCTS_FILE = path.join('storages', 'products.store.json');
```
Comment (VitalyaIlin) [inline shop/services.js:13]: You can create a separate file with constants to reuse them across the app

File: shop/services.js [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+
+uploadEvents.on('fileUploadStart', async () => await logEvent('File upload has started'));
+uploadEvents.on('fileUploadEnd', async () => await logEvent('File has been uploaded'));
+uploadEvents.on('fileUploadFailed', async (error) => await logEvent(`Error occurred, file upload failed: ${error}`));
+
+export const registrationUser = async (req, res) => {
+    const newUser = req.body
+
+    await userValidationSchema.validateAsync(newUser).catch(error => {
+        throw new ApiError(!newUser.name || !newUser.email || !newUser.password ? 402 : 400, error.details[0].message);
```
Comment (VitalyaIlin) [inline shop/services.js:40]: I don't think that 402 status code is necessary here
[https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/402](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/402  )

File: shop/services.js [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+            userId: userId,
+            products: []
+        }
+        carts.push(userCart);
+    }
+    userCart.products.push(product);
+    res.status(201).json(userCart);
+}
+
+export const deleteProd = (req, res) => {
```
Comment (VitalyaIlin) [inline shop/services.js:85]: In my opinion, `deleteProduct` or `removeProduct` are better names

**Review by VitalyaIlin — COMMENTED** [review]
check comments please

File: shop/services.js (hunk 4 above)
Comment (Kitsune1904) [reply shop/services.js:85]: Thanks I will change all

**Pull request merged**